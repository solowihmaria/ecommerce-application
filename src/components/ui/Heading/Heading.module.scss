.heading {
    margin: 0 0 1rem;
    font-family: inherit;
    font-weight: 600;
    line-height: 1.2;
    color: inherit;

    &--h1 {
        font-size: 2.5rem;
    }

    &--h2 {
        font-size: 2rem;
    }

    &--h3 {
        font-size: 1.75rem;
    }

    &--h4 {
        font-size: 1.5rem;
    }

    &--h5 {
        font-size: 1.25rem;
    }

    &--h6 {
        font-size: 1rem;
        font-weight: 700;
    }
}
