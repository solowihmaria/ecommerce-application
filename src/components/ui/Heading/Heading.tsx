import React from 'react';
import styles from './Heading.module.scss';
import type { HeadingProps } from './Heading.types';

export const Heading = ({
    level = 'h1',
    className = '',
    children,
    ...props
}: HeadingProps): React.JSX.Element => {
    const Tag = level;

    return (
        <Tag
            className={`${styles.heading} ${styles[`heading--${level}`]} ${className}`}
            {...props}
        >
            {children}
        </Tag>
    );
};
