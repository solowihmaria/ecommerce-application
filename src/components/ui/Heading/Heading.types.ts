import type { HTMLAttributes } from 'react';

type HeadingLevel = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';

export interface HeadingProps extends HTMLAttributes<HTMLHeadingElement> {
    level?: HeadingLevel;
}
