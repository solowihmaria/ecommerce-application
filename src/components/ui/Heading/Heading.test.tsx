import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Heading } from './Heading';
import type { HeadingProps } from './Heading.types';

describe('<Heading/> component', () => {
    test('Render Heading component without any properties', () => {
        render(<Heading>heading text</Heading>);
        const heading = screen.getByText(/heading text/i);
        expect(heading).toBeInTheDocument();
        expect(heading).toMatchSnapshot();
    });

    test('Render Heading component with properties', () => {
        const properties: HeadingProps = {
            level: 'h2',
            className: 'classExample',
            children: 'HEADING2',
        };
        render(<Heading {...properties} />);
        const heading = screen.getByText(/heading2/i);
        expect(heading).toBeInTheDocument();
        expect(heading).toHaveClass('classExample');
        expect(heading).toMatchSnapshot();
    });
});
