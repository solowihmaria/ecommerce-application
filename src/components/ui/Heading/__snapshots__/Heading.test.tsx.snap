// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> component Render Heading component with properties 1`] = `
<h2
  class="heading heading--h2 classExample"
>
  HEADING2
</h2>
`;

exports[`<Heading/> component Render Heading component without any properties 1`] = `
<h1
  class="heading heading--h1"
>
  heading text
</h1>
`;
